low-level/framework/utils/sysfs_interface.py

```diff
@@ -138,7 +138,23 @@ def fetch_nw_cable_status(self, nw_interface_path, interface):
                 carrier_indicator = cFile.read().strip()
             if carrier_indicator not in self.nw_phy_link_state.keys():
                 carrier_indicator = 'unknown'
-        except OSError as os_err:
-            return os_err.errno
+        except Exception:
+            carrier_indicator = 'unknown'
```

Comment:
About exception handling, @cdeshmukh  what is the expected behaviour if any exception occurs for eg. FileNotFound, should we through an exception or return empty or in this case 'unknown' as result?

Reply:
>About exception handling, @cdeshmukh what is the expected behaviour if any exception occurs??

Ideally generic exception should be at one place for entire lib/module, like we have in sspl_ll_d, when code executing in sspl service context. 
But here now, this code will get executed in Node CLI or HA context, where we need to have generic exception also, but dont avoid specific exception handling, which we have already in place, keep that too

>for eg. FileNotFound, should we through an exception or return empty or in this case 'unknown' as result?

You may do this in case of specific/generic exception too

Reply:
removed exception handling from here. and catching it in consumer code.